hack/run-e2e.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+GOPATH="${GOPATH:-$HOME/go}"
+
+echo "Checking cluster"
+kubectl config current-context
+JSONPATH='{range .items[*]}{@.metadata.name}{"\t"}Ready={@.status.conditions[?(@.type=="Ready")].status}{"\n"}{end}'
+if kubectl get nodes -o jsonpath="$JSONPATH" | grep "Ready=False"; then 
+    exit 1
+fi
+
+echo; echo "Checking helm and tiller install"
+helm version
+
+echo; echo "Checking for other pre-reqs"
+which jq        # Used to format some command output
+which docker
+
+echo; echo "Checking that the Metrics Server is deployed"
+kubectl get --raw "/apis/metrics.k8s.io/v1beta1/nodes" | jq .
+
+# If the script fails on error here, the deployment won't be cleaned up properly
+set +o errexit 
+
+echo; echo "Running deployment scripts"
+cd $GOPATH/src/github.com/Azure/azure-k8s-metrics-adapter/.azdevops/2_deploy
```

Comment:
move /.azdevops/2_deploy scripts to the hack folder.

Reply:
consider using use `pushd` and `pop` over `cd`

Reply:
From some brief googling it looks like the main plus of `pushd`/`popd` over `cd` is that it allows for more efficient file system navigation. Since it's being done in a script and not by a user, I will leave it as `cd` as it shouldn't matter in this case.